pmlc/rt/device.h

```diff
@@ -17,7 +18,8 @@ class Device {
 
   virtual std::unique_ptr<Executable>
   compile(const std::shared_ptr<pmlc::compiler::Program> &program,
-          mlir::ArrayRef<void *> bufptrs) = 0;
+          mlir::ArrayRef<util::BufferPtr> inputBuffers,
+          mlir::ArrayRef<util::BufferPtr> outputBuffers) = 0;
```

Comment:
@earhart Passing `BufferPtr` and splitting into inputs vs outputs makes some of the logic in other places a little easier. WDYT?

Reply:
Completely fine; we just need a convention for ordering.  It might also be good to talk somewhere about using the same buffer as an input and an output (explicitly noting that it's legal, if it's legal).  It'd also be useful to have another input for "things we expect to not change" -- I'm not quite sure what the plan is for constant folding.

Reply:
Ordering is determined by the `Program` (and actually by the end-user).

Reply:
I meant for the network function call -- "inputs before outputs".  And now I see the constants -- awesome!